ui/raidboss/data/06-ew/trial/zodiark-ex.ts

```diff
@@ -1,23 +1,297 @@
+  south: Outputs.south,
+  southwest: Outputs.southwest,
+  east: Outputs.east,
+  combo: {
+    en: '${first} / ${second}',
+  },
+};
+
+const fetchCombatantsByTargetID = async (targetId: string[]) => {
+  const decIds = [];
+  for (const id of targetId)
+    decIds.push(parseInt(id, 16));
+  const callData = await callOverlayHandler({
+    call: 'getCombatants',
+    ids: decIds,
+  });
+  return callData.combatants;
+};
 
 const triggerSet: TriggerSet<Data> = {
   zoneId: ZoneId.TheMinstrelsBalladZodiarksFall,
   timelineFile: 'zodiark-ex.txt',
+  initData: () => ({
+    activeSigils: [],
+    activeFrontSigils: [],
+    paradeigmaCounter: 0,
+    adikiaCounter: 0,
+  }),
   triggers: [
     {
       id: 'ZodiarkEx Ania',
       type: 'StartsUsing',
-      netRegex: NetRegexes.startsUsing({ id: '67EF', source: 'Zodiark' }),
-      netRegexDe: NetRegexes.startsUsing({ id: '67EF', source: 'Zodiark' }),
-      netRegexFr: NetRegexes.startsUsing({ id: '67EF', source: 'Zordiarche' }),
-      netRegexJa: NetRegexes.startsUsing({ id: '67EF', source: 'ゾディアーク' }),
-      response: Responses.tankBuster(),
+      netRegex: NetRegexes.startsUsing({ id: '6B63', source: 'Zodiark' }),
+      response: Responses.tankBusterSwap(),
+    },
+    {
+      id: 'ZodiarkEx Paradeigma',
+      type: 'Ability',
+      netRegex: NetRegexes.ability({ id: '67BF', source: 'Zodiark', capture: false }),
+      alertText: (data, _matches, output) => {
+        ++data.paradeigmaCounter;
+        if (data.paradeigmaCounter === 1)
+          return output.underQuetz!();
+      },
+      outputStrings: {
+        underQuetz: {
+          en: 'Under NW Quetzalcoatl',
+        },
+      },
+    },
+    {
+      id: 'ZodiarkEx Styx',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '67F3', source: 'Zodiark', capture: false }),
+      alertText: (_data, _matches, output) => output.text!(),
+      outputStrings: {
+        text: {
+          en: 'Stack x6',
+        },
+      },
+    },
+    {
+      // 67E4 Green Beam, 67E5 Rectangle, 67E6 Wedge
+      id: 'ZodiarkEx Arcane Sigil End',
+      type: 'Ability',
+      netRegex: NetRegexes.ability({ id: ['67E4', '67E5', '67E6'], source: 'Arcane Sigil' }),
+      run: (data, matches, _output) => {
+        for (let i = 0; i < data.activeSigils.length; ++i) {
+          const sig = data.activeSigils[i];
+          if (sig?.npcId === matches.sourceId)
+            data.activeSigils.splice(i, 1);
+        }
+      },
+    },
+    {
+      id: 'ZodiarkEx Blue Cone Tether',
+      type: 'Tether',
+      netRegex: NetRegexes.tether({ id: '00A4', source: 'Zodiark' }),
+      promise: async (data, matches) => {
+        const portalActors = await fetchCombatantsByTargetID([matches.targetId]);
+        for (const actor of portalActors) {
+          if (actor.ID) {
+            // console.log('blue ', { x: actor.PosX, y: actor.PosY, typeId: '67E6', npcId: actor.ID.toString(16).toUpperCase() });
```

Comment:
More debug logging to remove?

Reply:
Oh yeah forgot both of them after testing the north sigils